plugins/woocommerce-blocks/assets/js/data/payment/default-state.ts

```diff
@@ -14,6 +14,79 @@ import {
 import { SavedPaymentMethod } from './types';
 import { STATUS as PAYMENT_STATUS } from './constants';
 
+function getDefaultPaymentMethod() {
+	const defaultPaymentMethod = getSetting< SavedPaymentMethod | string >(
+		'defaultPaymentMethod',
+		''
+	);
+
+	if ( ! defaultPaymentMethod ) {
+		return '';
+	}
+
+	// defaultPaymentMethod is a string if a regular payment method is set.
+	if ( typeof defaultPaymentMethod === 'string' ) {
+		return defaultPaymentMethod;
+	}
+
+	// defaultPaymentMethod is a SavedPaymentMethod object if a saved payment method is returned.
+	if (
+		defaultPaymentMethod?.method?.gateway &&
+		defaultPaymentMethod?.tokenId
+	) {
+		return defaultPaymentMethod.method.gateway;
+	}
+
+	return '';
+}
+
+/**
+ * Set the default payment method data. This can be in two places,
+ * Either as part of the `defaultPaymentMethod` object or
+ * as a token stored in `wcSettings`.
+ */
+function getDefaultPaymentMethodData() {
```

Comment:
We need this when the default payment method is a saved payment method. It's either extracted from `defaultPaymentMethod` when it's an object, or from the `defaultPaymentMethodToken` setting

Reply:
This is so that the persisted payment method is loaded as the initial state, so we don't need to fetch it again from client. If it's a saved payment method it will also need the `paymentMethodData` object to be set.

Reply:
Doesn't _have_ to be before render, could be hydrated to the checkout store - but that maps to the checkout storeApi endpoint, so I thought this was a better approach